From: Guilhem Moulin <guilhem@debian.org>
Date: Fri, 5 Jun 2020 03:11:21 +0200
Subject: [PATCH] Make getnameinfo(3) errors non-fatal in report_sock()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

report_sock() is used to show the peer's address/name and port when the
‘-v’ flag is set.  Reverse resolution errors need not be fatal.

Bug-Debian: https://bugs.debian.org/961378

Imported from netcat-openbsd_1.228-1.debian.tar.xz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 netcat.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/netcat.c b/netcat.c
index 509b01fc935f..14264eb7136f 100644
--- a/netcat.c
+++ b/netcat.c
@@ -2220,9 +2220,11 @@ report_sock(const char *msg, const struct sockaddr *sa, socklen_t salen,
 	case 0:
 		break;
 	case EAI_SYSTEM:
-		err(1, "getnameinfo");
+		warn("getnameinfo");
+		return;
 	default:
-		errx(1, "getnameinfo: %s", gai_strerror(herr));
+		warnx("getnameinfo: %s", gai_strerror(herr));
+		return;
 	}
 
 	fprintf(stderr, "%s on %s %s\n", msg, host, port);
